src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,202 @@
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("The index passed must be greater or equal to 0 "
+                    + "and less or equal to" + size);
+        }
+        if (size - index == 0 || size == 0) {
+            add(value);
+            return;
+        }
+        iterateCurrentNode(index);
+        addNodeInTheMiddle(value);
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        verifyIndex(index);
+        iterateCurrentNode(index);
+        return currentNode.item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        if (size == 0) {
+            throw new IndexOutOfBoundsException("The list is empty");
+        }
+        verifyIndex(index);
+        iterateCurrentNode(index);
+        T returnItem = currentNode.item;
+        currentNode.item = value;
+        return returnItem;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        verifyIndex(index);
+        if (index == 0) {
+            return removeFirstNode();
+        }
+        iterateCurrentNode(index);
+        assignPreviousAndNextNode();
+        unlink(currentNode);
+        size--;
+        return currentNode.item;
     }
 
     @Override
     public boolean remove(T object) {
-        return false;
+        if (size == 0) {
+            throw new NullPointerException("The list is empty");
+        }
+        if (object == null ? head.item == null : object.equals(head.item)) {
+            return removeFromTheHead();
+        }
+        findInTheMiddle(object);
+        if (currentNode == null) {
+            return false;
+        }
+        assignPreviousAndNextNode();
+        unlink(currentNode);
+        size--;
+        return true;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private void iterateCurrentNode(int index) {
+        currentNode = head;
+        for (int i = 0; i < index; i++) {
+            currentNode = currentNode.next;
+        }
+    }
+
+    private void verifyIndex(int index) {
+        if (index < 0 || index > size - 1) {
+            throw new IndexOutOfBoundsException("The index passed must be equal "
+                    + "or greater than 0 and less than " + size);
+        }
+    }
+
+    private void addNodeInTheMiddle(T value) {
+        previousNode = currentNode.prev;
+        Node<T> newElement = new Node<>(value);
+        newElement.prev = previousNode;
+        newElement.next = currentNode;
+        if (currentNode == head || currentNode != null && currentNode.equals(head)) {
```

Comment:
The condition 'currentNode == head || currentNode != null && currentNode.equals(head)' is redundant. 'currentNode == head' is sufficient.